recog/src/main/java/com/rapid7/recog/provider/RecogMatchersProvider.java

```diff
@@ -113,8 +113,12 @@ private void parseFromWalkablePath(Path path) {
           try (Reader reader = Files.newBufferedReader(file)) {
             int extIndex = fileName.lastIndexOf(".xml");
             RecogMatchers matchers = parser.parse(reader, extIndex > 0 ? fileName.substring(0, extIndex) : fileName);
-            matchersByFileName.put(fileName, matchers);
-            matchersByKey.put(matchers.getKey(), matchers);
+            if (matchers != null) {
```

Comment:
After thinking about this further, instead of adding new conditional logic should would it make more sense to have `parse` throw a `ParseException` if the end result is `null`?

Reply:
Probably, yes. I'm happy to come back and refactor this further (should be tomorrow) if you're ok with this as an interim fix?
We'd like to get customers back to scanning asap